utils/frame/try-runtime/remote-externalities/src/lib.rs

```diff
@@ -0,0 +1,494 @@
+	}
+
+	/// Build `Self` from a network node denoted by `uri`.
+	async fn load_remote(&self) -> Vec<KeyPair> {
+		let config = self.as_online();
+		let at = self.as_online().at.unwrap().clone();
+		info!(target: LOG_TARGET, "scraping keypairs from remote node {} @ {:?}", config.uri, at);
+
+		let keys_and_values = if config.modules.len() > 0 {
+			let mut filtered_kv = vec![];
+			for f in config.modules.iter() {
+				let hashed_prefix = StorageKey(twox_128(f.as_bytes()).to_vec());
+				let module_kv = self.rpc_get_pairs(hashed_prefix.clone(), at).await;
+				info!(
+					target: LOG_TARGET,
+					"downloaded data for module {} (count: {} / prefix: {:?}).",
+					f,
+					module_kv.len(),
+					hashed_prefix.hex_display(),
+				);
+				filtered_kv.extend(module_kv);
+			}
+			filtered_kv
+		} else {
+			info!(target: LOG_TARGET, "downloading data for all modules.");
+			self.rpc_get_pairs(StorageKey(vec![]), at).await.into_iter().collect::<Vec<_>>()
+		};
+
+		keys_and_values
+	}
+
+	async fn init_remote_client(&mut self) {
+		self.as_online_mut().at = Some(self.rpc_get_head().await);
+		self.chain = self.chain_name().await;
+	}
+
+	async fn pre_build(mut self) -> Vec<KeyPair> {
+		let mut base_kv = match self.mode.clone() {
+			Mode::Offline(config) => self.load_cache(&config.cache.path()),
+			Mode::Online(config) => {
+				self.init_remote_client().await;
+				let kp = self.load_remote().await;
+				if let Some(c) = config.cache {
+					self.save_cache(&kp, &c.path());
+				}
+				kp
+			}
+		};
+
+		info!(
+			target: LOG_TARGET,
+			"extending externalities with {} manually injected keys",
+			self.inject.len()
+		);
+		base_kv.extend(self.inject.clone());
+		base_kv
+	}
+}
+
+// Public methods
+impl Builder {
+	/// Create a new builder.
+	pub fn new() -> Self {
+		Default::default()
+	}
+
+	/// Inject a manual list of key and values to the storage.
+	pub fn inject(mut self, injections: &[KeyPair]) -> Self {
+		for i in injections {
+			self.inject.push(i.clone());
+		}
+		self
+	}
+
+	/// Configure a cache to be used.
+	pub fn mode(mut self, mode: Mode) -> Self {
+		self.mode = mode;
+		self
+	}
+
+	/// Build the test externalities.
+	pub async fn build(self) -> TestExternalities {
+		let kv = self.pre_build().await;
+		let mut ext = TestExternalities::new_empty();
+
+		info!(target: LOG_TARGET, "injecting a total of {} keys", kv.len());
+		for (k, v) in kv {
+			let (k, v) = (k.0, v.0);
+			ext.insert(k, v);
+		}
+		ext
+	}
+}
+
+#[cfg(test)]
+mod tests {
+	use super::*;
+
+	#[async_std::test]
+	#[cfg(feature = "remote-test")]
+	async fn can_build_one_pallet() {
+		let _ = env_logger::Builder::from_default_env()
+			.format_module_path(false)
+			.format_level(true)
+			.try_init();
+
+		Builder::new()
+			.mode(Mode::Online(OnlineConfig {
+				modules: vec!["Proxy".into()],
+				..Default::default()
+			}))
+			.build()
+			.await
+			.execute_with(|| {});
+	}
+
+	#[async_std::test]
+	async fn can_load_cache() {
```

Comment:
that binary file (20k) that I bushed will allow us to have at least this one test for this crate.. not sure what to do about it.